### DISCUSSION TITLE
Re-usable Components Library

### DISCUSSION THREAD

**Discussion by AdsonCicilioti** [unocss/unocss#2660] [Q&A]
Hi,

 I'm looking for a tool that allows me to create reusable components.  At first in React.
 An important premise would be the possibility of using pre-compiled styles but also changing them at runtime.
 And another very important thing would be the cleaning of unused styles, because a big challenge of Component Libs with pre-compiled styles is that they already bring a complete Bundle of all library components.  Leading to the need for some kind of Wrapper Provider that scans which components were used in the application or page.

 I know that these are early questions and concerns and the responsibility of the build of the application that will use this Component Lib, but beforehand I bring this question that I would also like to know if UnoCSS covers it and how or what strategy would be more appropriate.

**Comment by siduck** [comment]
@AdsonCicilioti did you manage to make it?
